napari/layers/utils/text.py

```diff
@@ -39,74 +44,59 @@ class TextManager:
         The default value is 'translucent'
     visible : bool
         Set to true of the text should be displayed.
-
-    Attributes
-    ----------
-    text : array
-        the strings to be displayed
-    rotation : float
-        Angle of the text elements around the anchor point. Default value is 0.
-    anchor : str
-        The location of the text origin relative to the bounding box.
-        Should be 'center', 'upper_left', 'upper_right', 'lower_left', or 'lower_right'.
-    translation : np.ndarray
-        Offset from the anchor point.
-    color : array
-        Font color for the text
-    size : float
-        Font size of the text. Default value is 12.
-    The blending mode that determines how RGB and
-        alpha values of the layer visual get mixed. Allowed values are
-        {'opaque', 'translucent', and 'additive'}. Note that 'opaque` blending
-        is not recommended, as colors the bounding box surrounding the text.
-    visible : bool
-        Set to true of the text should be displayed.
     """
 
-    def __init__(
-        self,
-        text,
-        n_text,
-        properties={},
-        rotation=0,
-        translation=0,
-        anchor='center',
-        color='cyan',
-        size=12,
-        blending='translucent',
-        visible=True,
-    ):
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            text=Event,
-            rotation=Event,
-            translation=Event,
-            anchor=Event,
-            color=Event,
-            size=Event,
-            blending=Event,
-            visible=Event,
-        )
-
-        self.events.block_all()
-        self._rotation = rotation
-        self._anchor = Anchor(anchor)
-        self._translation = translation
-        self._color = transform_color(color)[0]
-        self._size = size
-        self._blending = self._check_blending_mode(blending)
-        self._visible = visible
-
-        self._set_text(text, n_text, properties)
-        self.events.unblock_all()
+    text: Union[np.ndarray, str]
```

Comment:
I am sorry if I missed this in some of the previous threads, but are the events associated with these properties emitted when they are set in the `__init__()` method?

Reply:
no apology necessary... all of this needs some nice documentation to go with it to introduce the new patterns without needing to go through all of the threads!
Nope, events will not be emitted during init. (but, now that you mention it, I don't think I have an explicit test for that... so will add)

Reply:
actually... is that the behavior you were hoping for?  I noticed that you had that event blocker in the `__init__` before ...

Reply:
> actually... is that the behavior you were hoping for? I noticed that you had that event blocker in the __init__ before ...

I think that most of the time, we don't want to emit the property change events during the `__init__`since those are often connected to some sort of refresh. I think if there are ever cases where that's required, we can add it explicitly in the `__post_init__` right?